Library/ViewModels/ProjectCreatorViewModel.swift

```diff
@@ -66,10 +69,22 @@ ProjectCreatorViewModelOutputs {
       .filter { _ in AppEnvironment.current.currentUser != nil }
       .map { (MessageSubject.project($0), .projectPage) }
 
-    self.goToSafariBrowser = navigationAction
-      .filter { $0.navigationType == .linkActivated }
-      .filter { !isMessageCreator(request: $0.request) }
-      .map { $0.request.url }
+    self.goBackToProject = Signal.combineLatest(project, navigationAction)
+      .filter { $1.navigationType == .linkActivated }
+      .filter { (arg: (Project, WKNavigationActionData)) -> Bool in
```

Comment:
Sometimes Xcode complains about a tuple not supporting destructuring and then one has to do this but you'll probably find that once it can infer your closure's return type correctly then you can probably remove this and just do:

```
.filter { project, navigation in
  project.urls.web.project != navigation.request.url?.absoluteString
}
```

Either is fine - just mentioning for interest.